crates/astria-sequencer-types/src/sequencer_block_data.rs

```diff
@@ -42,14 +46,19 @@ pub struct RollupData {
 
 /// `SequencerBlockData` represents a sequencer block's data
 /// to be submitted to the DA layer.
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
```

Comment:
serde provides this attribute: https://serde.rs/container-attrs.html#try_from

Can you implement a `TryFrom<RawSequencerBlockData>` (I'd prefer to not impl a single-use `TryFrom` and instead use a custom deserializer, but in this case it's probably fine).

This way you can still deserialize, but in addition you get verification. Right now you still serialize directly into a `SequencerBlockData` and have no assurance that it's legitimate (invariants upheld, etc).

Also see https://serde.rs/container-attrs.html#into for the inverse

Reply:
nice, added!